test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetConsoleTestCase.cs

```diff
@@ -682,6 +682,41 @@ public async Task UpdatePackageForPR_PackageNamespace_WithMultipleFeedsWithIdent
             solutionService.Save();
         }
 
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetPackagesConfigTemplates))]
+        public async Task VerifyCatchFileInsideObjFolder(ProjectTemplate projectTemplate)
+        {
+            using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger))
+            {
+                //Arrange
+                var packageName = "TestPackage";
+                var packageVersion = "1.0.0";
+                await CommonUtility.CreatePackageInSourceAsync(testContext.PackageSource, packageName, packageVersion);
+                var project = testContext.SolutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreConsoleApp, "NetCore");
+                var solutionService = VisualStudio.Get<SolutionService>();
+                var nugetConsole = GetConsole(testContext.Project);
+                nugetConsole.InstallPackageFromPMC(packageName, packageVersion);
+                FileInfo CatchFilePath = CommonUtility.GetCatchFilePath(project.FullPath);
+
+                //Act
+                testContext.SolutionService.Build();
+                testContext.NuGetApexTestService.WaitForAutoRestore();
+                CommonUtility.WaitForFileExists(CatchFilePath);
+
+                project.Rebuild();
+                CommonUtility.WaitForFileExists(CatchFilePath);
+
+                project.Clean();
+                CommonUtility.WaitForFileNotExists(CatchFilePath);
+
+                nugetConsole.Execute("dotnet restore");
```

Comment:
Can we skip this step? `dotnet restore` tests are separate from this. 

Plus, there's no guarantee that dotnet restore and VS restore contain the same code.

Reply:
Hi @nkolev92, we are not expect to test project restoring in this test case. The test purpose of executing `dotnet restore` (as below step5) is to see if the cache file can get back after cleaning solution in previous step. Corresponding manual test case is here:

**Steps:** 
1.Create above project and install corresponding PackageReference package.
2.Build solution, observe there should generate a restore cache file inside obj/ folder.
3.Rebuild solution, observe there should generate a restore cache file inside obj/ folder.
4.Clean solution, observe there should clean the restore cache file inside obj/ folder.
5.Restore NuGet Package, observe there should also generate or persist this restore cache file inside obj/ folder. 
**Expect Result:** 
There should generate a restore cache file inside obj/ folder in step2 & step3 & step5, and no cache file inside obj/ folder in step4. 

Do you mean the `dotnet restore` would not work for above scenario? If it doesn't work, should we use VS restore? (We used `dotnet restore` instead of VS restore since that's easier.) 

If all the restore should be skipped in the test case, we can do that. Have any existed automation test case covered above step5 (verifying cache file generated after doing restore)?

Reply:
> Hi @nkolev92, we are not expect to test project restoring in this test case. The test purpose of executing dotnet restore (as below step5) is to see if the cache file can get back after cleaning solution in previous step. Corresponding manual test case is here:

Restore is what generates this file. So we're testing restore. By calling dotnet restore, you're testing dotnet restore since it's what's creating the file. 

> Do you mean the dotnet restore would not work for above scenario? If it doesn't work, should we use VS restore? (We used dotnet restore instead of VS restore since that's easier.)

I think it's more appropriate to call Visual Studio restore. 

> If all the restore should be skipped in the test case, we can do that. Have any existed automation test case covered above step5 (verifying cache file generated after doing restore)?

We actually have tests that verify the behavior. https://github.com/NuGet/NuGet.Client/blob/ac7977c16f6d4c7d48ad3c624b6d5caa46c4447f/test/EndToEnd/tests/NetCoreProjectTest.ps1#L141. So maybe this isn't as valuable. 
https://github.com/NuGet/NuGet.Client/blob/ac7977c16f6d4c7d48ad3c624b6d5caa46c4447f/test/EndToEnd/tests/NetCoreProjectTest.ps1#L106-L120 has a test that clean deletes the cache file. 

I think the test here should call VS restore and check that or just remove it as duplicate of the above 2.

Reply:
OK, we will remove the `dotnet restore` step from this test case. Thanks.